org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java

```diff
@@ -343,7 +344,7 @@ private void deriveEnums(EnumRule rule) {
 						if (nodes.size() > 1)
 							throw new IllegalStateException("Unexpected nodes found: " + nodes);
 						INode node = nodes.get(0);
-						String text = node.getText().trim().replace("^", "");
+						String text = Joiner.on("").join(Iterables.transform(Iterables.filter(node.getLeafNodes(), e -> !e.isHidden()), e -> e.getText().trim())).trim().replace("^", "");
```

Comment:
@szarnekow  do we have a util for something like this?

Reply:
I think this is what `org.eclipse.xtext.nodemodel.util.NodeModelUtils.getTokenText(INode)` does.